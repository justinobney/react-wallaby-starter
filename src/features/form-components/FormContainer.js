import React from 'react';
import {Container, Dimmer, Loader} from 'semantic-ui-react';
import styled from 'react-emotion';
import P from 'prop-types';

const DimmableWrapper = styled(Dimmer.Dimmable)`
  box-shadow: 0 0 15px 2px rgba(0, 0, 0, 0.1);
  border-radius: 5px;
  border: solid 1px rgba(0, 0, 0, 0.1);
  padding: 20px;
  background: #fff;
  margin-bottom: 10px;
`;

const FormContainer = ({children, loading, ...props}) => {
  return (
    <DimmableWrapper as={Container} dimmed={loading} {...props}>
      {children}
      <Dimmer active={loading} inverted>
        <Loader>Loading...</Loader>
      </Dimmer>
    </DimmableWrapper>
  );
};

FormContainer.propTypes = {
  children: P.node.isRequired,
  loading: P.bool,
};

export default FormContainer;
