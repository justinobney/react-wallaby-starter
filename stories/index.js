import React from 'react';
import {storiesOf, linkTo} from '@kadira/storybook';
import Welcome from './Welcome';

import 'semantic-ui-css/semantic.css';

storiesOf('Welcome', module).add('to Storybook', () => (
  <Welcome showApp={linkTo('Button')} />
));
