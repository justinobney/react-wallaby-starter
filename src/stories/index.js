import React from 'react';

import {storiesOf} from '@storybook/react';
// import {action} from '@storybook/addon-actions';
import {linkTo} from '@storybook/addon-links';
import {withInfo} from '@storybook/addon-info';

import {Welcome} from '@storybook/react/demo';
import FormContainer from '../features/form-components/FormContainer';
import '../styles/imports';

storiesOf('Welcome', module).add('to Storybook', () => (
  <Welcome showApp={linkTo('FormContainer')} />
));

storiesOf('FormContainer', module)
  .add(
    'Default',
    withInfo('Given only children')(() => (
      <FormContainer>Some Content</FormContainer>
    ))
  )
  .add(
    'Loading',
    withInfo('Dim the content by passing a loading property')(() => (
      <FormContainer loading>My content is dimmed</FormContainer>
    ))
  );
