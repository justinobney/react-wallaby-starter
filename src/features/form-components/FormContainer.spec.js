// import React from 'react';
import {create} from 'react-test-renderer';
import {createSerializer} from 'jest-emotion';
import * as emotion from 'emotion';

// import FormContainer from './FormContainer';
import {stories} from './FormContainer.stories';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Component: FormContainer', () => {
  describe('Stories', () => {
    Object.keys(stories).map(story => {
      it(story, () => {
        const tree = create(stories[story]).toJSON(); // eslint-disable-line
        expect(tree).toMatchSnapshot();
      });
      return undefined;
    });
  });
});
