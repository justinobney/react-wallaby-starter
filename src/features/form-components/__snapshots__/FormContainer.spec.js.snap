// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormContainer Stories Default 1`] = `
.emotion-0 {
  box-shadow: 0 0 15px 2px rgba(0,0,0,0.1);
  border-radius: 5px;
  border: solid 1px rgba(0,0,0,0.1);
  padding: 20px;
  background: #fff;
  margin-bottom: 10px;
}

<div
  className="dimmable emotion-0"
>
  Some Content
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui text loader"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: FormContainer Stories Loading 1`] = `
.emotion-0 {
  box-shadow: 0 0 15px 2px rgba(0,0,0,0.1);
  border-radius: 5px;
  border: solid 1px rgba(0,0,0,0.1);
  padding: 20px;
  background: #fff;
  margin-bottom: 10px;
}

<div
  className="dimmed dimmable emotion-0"
>
  My content is dimmed
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui text loader"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: FormContainer Stories Loading Message 1`] = `
.emotion-0 {
  box-shadow: 0 0 15px 2px rgba(0,0,0,0.1);
  border-radius: 5px;
  border: solid 1px rgba(0,0,0,0.1);
  padding: 20px;
  background: #fff;
  margin-bottom: 10px;
}

<div
  className="dimmed dimmable emotion-0"
>
  My content is dimmed
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui text loader"
        >
          Custom Loading Message
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Component: FormContainer Stories Passes Extra Props 1`] = `
.emotion-0 {
  box-shadow: 0 0 15px 2px rgba(0,0,0,0.1);
  border-radius: 5px;
  border: solid 1px rgba(0,0,0,0.1);
  padding: 20px;
  background: #fff;
  margin-bottom: 10px;
}

<form
  action={undefined}
  className="ui form dimmable emotion-0"
  onSubmit={[Function]}
>
  Some Content
  <div
    className="ui inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui text loader"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</form>
`;
